icepyx/core/granules.py

```diff
@@ -178,18 +177,18 @@ def get_avail(self, CMRparams, reqparams):
             )
 
             results = json.loads(response.content)
-
-            # print(results)
-
+            
             try:
-                if not results["feed"]["entry"]:
-                    # Out of results, so break out of loop
-                    break
-            except KeyError:
-                if "errors" in results.keys():
-                    raise QueryError.NsidcQueryError(
-                        str(results["errors"])
-                    )
+                response.raise_for_status()
+            except requests.HTTPError as e:
+                if b'errors' in response.content:  # If CMR returns a bad status with extra information, display that
+                    raise icepyx.core.exceptions.NsidcQueryError(str(results["errors"])) # exception chaining will display original exception too 
+                else:  # If no 'errors' key, just reraise original exception
+                    raise
```

Comment:
If you move this bit before `results = ...` instead (and use `response.content` instead of `str(results["errors"])`), I think it is a bit safer/simpler. For example, with some errors like a gateway timeout or server problem, the response content may not be json parse-able, but the status code would still indicate an error. As it is, in that case, the user would get a `JSONParseError` instead of an `HTTPError`.

I hope I am not being too much of a pain...you will probably never request a review from me again :grimacing: 

:)

Reply:
Haha - never fear, you're not off the hook yet. I'm using on the other side of PR reviews, so it's nice to have some feedback and I'm learning a bunch!

What you're saying makes sense. I'll move it.